mesonbuild/scripts/gtkdochelper.py

```diff
@@ -95,6 +96,7 @@ def build_gtkdoc(source_root, build_root, doc_subdir, src_subdirs,
     doc_src = os.path.join(source_root, doc_subdir)
     abs_out = os.path.join(build_root, doc_subdir)
     htmldir = os.path.join(abs_out, 'html')
+    final_destination = os.path.join(install_prefix, datadir, install_dir)
```

Comment:
Actually, reading the glib example, it should be just prefix and datadir.

Reply:
Now I remember why I passed the full path instead of just the path to the `html` directory. It becomes problematic when a project passes a weird absolute `install_dir` to `gnome.gtkdoc` (though maybe we can hope no one does this?)

Reply:
I guess that is one benefit of string concatenation https://gitlab.gnome.org/GNOME/gtk-doc/-/blob/250a21f24365b6fed6271b5f235307d9fbef5eeb/buildsystems/autotools/gtk-doc.make#L42, it does not allow projects to install docs to weird locations, making it completely up to distros.

I guess it is too late for disallowing passing absolute values not under `${datadir}/share/gtk-doc/html` to `install_dir`?